### PR TITLE
Document analog IO

### PR DESCRIPTION
**Pull request by jonnew**
- First shot at using XML documentaiton to generate coherent and user-friendly documentation

### CODE REVIEW

File: OpenEphys.Onix/OpenEphys.Onix/ConfigureAnalogIO.cs [hunk 1]
```diff
@@ -4,115 +4,204 @@
...
         public AnalogIODirection Direction10 { get; set; }
 
+        /// <summary>
+        /// Get or set the direction of channel 11.
+        /// </summary>
         [Category(AcquisitionCategory)]
         [Description("The direction of channel 11.")]
         public AnalogIODirection Direction11 { get; set; }
 
+        /// <summary>
```
[the remark this answers is not in the document]
Comment (jonnew) [reply OpenEphys.Onix/OpenEphys.Onix/ConfigureAnalogIO.cs:200]: For param and return look at example.

File: OpenEphys.Onix/OpenEphys.Onix/ConfigureAnalogIO.cs [hunk 2]
```diff
@@ -244,25 +333,55 @@ public static float GetVoltsPerDivision(AnalogIOVoltageRange voltageRange)
...
     }
 
+    /// <summary>
+    /// Analog input ADC voltage range.
+    /// </summary>
     public enum AnalogIOVoltageRange
     {
-        [Description("+/-10.0 V")]
+        /// <summary>
+        /// +/-10.0 volts.
```
Comment (jonnew) [inline OpenEphys.Onix/OpenEphys.Onix/ConfigureAnalogIO.cs:342]: Use unicode for plus minus.

File: OpenEphys.Onix/OpenEphys.Onix/AnalogOutput.cs [hunk 3]
```diff
@@ -7,15 +7,33 @@
...
+        /// 16-bit integer. In this case, the output voltage always corresponds to <see cref="AnalogIOVoltageRange.TenVolts"/>.
+        /// When <see cref="AnalogIODataType.Volts"/> is selected, 32-bit floating point voltages between -10 and 10 volts are sent
+        /// directly to the DACs.
+        /// </remarks>
         public AnalogIODataType DataType { get; set; } = AnalogIODataType.S16;
 
+        /// <summary>
+        /// Send samples to analog outputs.
+        /// </summary>
+        /// <param name="source"> A sequence of 12xN sample matrices containing the analog data to write to channels 0 to 11.</param>
```
Comment (bparks13) [inline OpenEphys.Onix/OpenEphys.Onix/AnalogOutput.cs:35]: Should we specify the type in the param tag? e.g., `<see cref="Mat"/>` for this one?

File: OpenEphys.Onix/OpenEphys.Onix/ConfigureAnalogIO.cs [hunk 4]
```diff
@@ -4,115 +4,212 @@
...
             : base(typeof(AnalogIO))
         {
             DeviceAddress = 6;
         }
 
+        /// <summary>
+        /// Get or set the device enable state.
+        /// </summary>
+        /// <remarks>
+        /// If set to true, <see cref="AnalogInput"/> will produce data. If set to false, <see cref="AnalogInput"/> will not produce data.
```
Comment (bparks13) [inline OpenEphys.Onix/OpenEphys.Onix/ConfigureAnalogIO.cs:26]: If this is disabled, are we able to send `AnalogOutput` data? I would assume not, so we might want to make that explicit

File: OpenEphys.Onix/OpenEphys.Onix/ConfigureAnalogIO.cs [hunk 5]
```diff
@@ -4,115 +4,212 @@
...
+        /// <remarks>
+        /// This will schedule analog IO hardware configuration actions that can be applied by a <see cref="StartAcquisition"/> object prior to data collection.
+        /// </remarks>
+        /// <param name="source">
+        /// The sequence of <see cref="ContextTask"/> objects on which to apply the analog IO configuration.
+        /// </param>
+        /// <returns>
+        /// A sequence of <see cref="ContextTask"/> objects that is identical to <paramref name="source"/> in which each <see cref="ContextTask"/> has been instructed
+        /// to apply the analog IO configuration.
+        /// </returns>
```
Comment (bparks13) [inline OpenEphys.Onix/OpenEphys.Onix/ConfigureAnalogIO.cs:212]: Potentially changing this wording

File: OpenEphys.Onix/OpenEphys.Onix/ConfigureAnalogIO.cs [hunk 6]
```diff
@@ -244,25 +341,55 @@ public static float GetVoltsPerDivision(AnalogIOVoltageRange voltageRange)
...
     }
 
+    /// <summary>
+    /// Specifies the analog input ADC voltage range.
+    /// </summary>
     public enum AnalogIOVoltageRange
     {
-        [Description("+/-10.0 V")]
+        /// <summary>
+        /// ±10.0 volts.
```
Comment (bparks13) [inline OpenEphys.Onix/OpenEphys.Onix/ConfigureAnalogIO.cs:350]: Unicode looks great, nice and clean

File: OpenEphys.Onix/OpenEphys.Onix/ConfigureAnalogIO.cs [hunk 7]
```diff
@@ -244,25 +341,55 @@ public static float GetVoltsPerDivision(AnalogIOVoltageRange voltageRange)
...
+    /// Specifies analog channel direction.
+    /// </summary>
     public enum AnalogIODirection
     {
+        /// <summary>
+        /// Input to breakout board.
+        /// </summary>
         Input = 0,
+        /// <summary>
+        /// Output from breakout board with loopback.
```
Comment (bparks13) [inline OpenEphys.Onix/OpenEphys.Onix/ConfigureAnalogIO.cs:376]: What does 'loopback' mean in this instance?

**Review by bparks13 — COMMENTED** [review]
Overall looks good and is mostly consistent with how I have structured the Neuropixels docs in #144, I did have some questions that we should iron out before finalizing PRs

**Comment by bparks13** [comment]
Closing this PR, changes were merged and are being addressed in #162.

**Pull request closed without merging**